src/controllers/patient.controller.js

```diff
@@ -0,0 +1,101 @@
+const Patients = require("../models/patient.model")
+const Prescription = require("../models/prescription.model")
+const Bill = require("../models/bill.model")
+
+const bcrypt = require("bcrypt")
+
+const {newPatientSchema, prescriptionIdSchema} = require("../validators/patient.validate")
+
+exports.newPatient = async (req,res) => {
+    try{
+        const result = await newPatientSchema.validateAsync(req.body)
+        
+        const patient = new Patients({
+            patientName: result.patientName,
+            patientEmail: result.patientEmail,
+            patientPassword: await bcrypt.hash(result.patientPassword,10),
+            patientDob: result.patientDob,
+            patientGender: result.patientGender,
+            patientWeight: result.patientWeight,
+            patientAddress: result.patientAddress,
+            patientPhone: result.patientPhone,
+            
+        })
+        const newPatient = await patient.save()
+       return res.json({
+                            status : "Success",
+                            error : false,
+                            message : "Successfully register new patient User",
+                            data : newPatient
+                        });  
+    }
+    catch(err){
+        if(err.isJoi === true){
+            return res.json({
+                status : "Failed  validation",
+                error : true,
+                message : "Failed to register new patient user",
+                data : { result: err}
+            })
+        }
+        else{
+        return res.json({
+            status : "Failed",
+            error : true,
+            message : "Failed to register new patient User",
+            data : { result: err}
+                        })
+            }
+        }
+    }
+
+exports.prescription =  async (req,res) => {
+    try{
+        let prescription = await Prescription.find({patientId:res.current_user.patient._id})
```

Comment:
Here you lack validation and also the `res.current_user.patient._id` is incorrect, as the payload is received from the `req` and not from `res` secondly we have to strictly follow the camelCase we don't do snake case in js.

Reply:
changed to camelCase and explained situation related to `res.current_user` [here](https://github.com/Sankalp-Sachapara/HospitalPractice/pull/3/files#r1045548274)

Reply:
👍